scripts/1inch-callee-data-fetcher/index.js

```diff
@@ -0,0 +1,83 @@
+import path from 'path';
+import url from 'url';
+import { spawn } from 'child_process';
+import fetch from 'node-fetch';
+import { utils } from 'ethers';
+
+const CHAIN_ID = 1;
+const BASE_URL = `https://api.1inch.io/v5.0/${CHAIN_ID}`; // see https://docs.1inch.io/docs/aggregation-protocol/api/swagger/
+const EXPECTED_SIGNATURE = '0x12aa3caf'; // see https://www.4byte.directory/signatures/?bytes4_signature=0x12aa3caf
+
+async function executeOneinchRequest(methodName, queryParams) {
+    const url = `${BASE_URL}${methodName}?${new URLSearchParams(queryParams)}`;
+    const response = await fetch(url).then(res => res.json());
+    if (response.error) {
+        console.error(
+            `1inch API call to "${methodName}" failed with:`,
+            response.error,
+            'request parameters:',
+            swapParams
+        );
+        throw new Error(`1inch API call failed with "${response.error}"`);
+    }
+    return response;
+}
+
+async function getOneinchValidProtocols() {
+    // Fetch all supported protocols except for the limit orders
+    const response = await executeOneinchRequest('/liquidity-sources');
+    const protocolIds = response.protocols.map(protocol => protocol.id);
+    return protocolIds.filter(protocolId => !protocolId.toLowerCase().includes('limit'));
+}
+
+async function getOneinchSwapParameters(linkAmount) {
+    const swapParams = {
+        fromTokenAddress: '0x514910771AF9Ca656af840dff83E8264EcF986CA', // LINK
+        toTokenAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F', // MCD_DAI
+        fromAddress: '0x0000000000000000000000000000000000000000', // have to be callee address
+        amount: linkAmount,
+        slippage: 1,
+        allowPartialFill: false, // disable partial fill
+        disableEstimate: true, // disable allowance checks
+        compatibilityMode: true, // always receive parameters for the `swap` call
```

Comment:
do you know what is the meaning of the compatibility mode field?

Reply:
The `compatibilityMode` is not mentioned in the [documentation](https://docs.1inch.io/docs/aggregation-protocol/api/swap-params/), but present in the [swagger](https://docs.1inch.io/docs/aggregation-protocol/api/swagger/). If not set, the API provides calldata for different methods of the aggregation router (eg directly to the `uniswapV3Swap`). Receiving calldata in different structure depending on the platform, doesn't allow to parse and rewrite parameters in the contract